runtime/src/main/scala/akka/grpc/scaladsl/GrpcMarshalling.scala

```diff
@@ -19,7 +19,17 @@ import scala.concurrent.Future
 object GrpcMarshalling {
   def unmarshal[T](req: HttpRequest)(implicit u: ProtobufSerializer[T], mat: Materializer): Future[T] = {
     val messageEncoding = `Message-Encoding`.findIn(req.headers)
-    req.entity.dataBytes.via(Grpc.grpcFramingDecoder(messageEncoding)).map(u.deserialize).runWith(Sink.head)(mat)
+    implicit val ec = mat.executionContext
+    req.entity.dataBytes
+      .via(Grpc.grpcFramingDecoder(messageEncoding))
+      .map(u.deserialize)
+      .runWith(Sink.headOption)(mat)
+      .flatMap {
+        _ match {
+          case Some(element) => Future.successful(element)
+          case None          => Future.failed(new MissingParameterException())
```

Comment:
Does this only happen on a missing parameter, should it rather be something like "invalid request" rather because it could be anything with the encoding/serialized form? Or would that lead to a failed stream already rather?

Reply:
Yes, I would expect other errors to lead to a failed stream rather than an empty one